### ISSUE TITLE
Runtime error: Error running the bot: Failed to initialize browser: Message: unknown error: cannot find Chrome binary

### ISSUE THREAD

**Issue by lscalize** [feder-cr/linkedIn_auto_jobs_applier_with_AI#339]
Hi, could anyone help me with this issue?

![image](https://github.com/user-attachments/assets/91b050b1-0d4b-454d-8ee0-4fd0985a54c1)

**Comment by aimaanmansuri21** [comment]
I encountered this issue and what I did is simply deactivated the virtual environment and restarted my pc, it still kept happening, I repeated this process 3 times and on the third try, it just worked!

**Comment by lscalize** [comment]
I've been trying for a while, but it's still not working.

**Comment by lscalize** [comment]
@aimaanmansuri21 I'm trying to run it on Google Colab. Do you think that I need to try somewhere else?

**Comment by lscalize** [comment]
I've tried to solve it using chatgpt, but it did not work.

I've followed these steps below.

Could someone help me pls?

# Update system
!apt-get update

# Install Chromium and ChromeDriver
!apt-get install -y chromium-chromedriver
!apt-get install -y chromium

# Setup the Selenium to Use Chromium
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.chrome.options import Options

# Setup Chrome
chrome_options = Options()
chrome_options.add_argument("--headless")  # Modo headless
chrome_options.add_argument("--no-sandbox")
chrome_options.add_argument("--disable-dev-shm-usage")
chrome_options.add_argument("--disable-gpu")
chrome_options.add_argument("--remote-debugging-port=9222")

# Directory Chrome and ChromeDriver
chrome_options.binary_location = '/usr/bin/chromium-browser'
service = Service('/usr/lib/chromium-browser/chromedriver')

**Comment by feder-cr** [comment]
@lscalize i think you can use on Google collab, because colab don't have chrome

**Comment by lscalize** [comment]
@feder-cr so, what do you suggest? So I need to use the Jupyter notebook instead? Another tool?

**Comment by lscalize** [comment]
@feder-cr or is it possible to run it directly through github?

**Comment by Lumexio** [comment]
Same problem here
i'm currently on Ubuntu 24

and im using my own Resume
```
Runtime error: Error running the bot: Failed to initialize browser: Message: unknown error: cannot find Chrome binary
Stacktrace:
#0 0x562187fd84e3 <unknown>
#1 0x562187d07c76 <unknown>
#2 0x562187d2e757 <unknown>
#3 0x562187d2d029 <unknown>
#4 0x562187d6bccc <unknown>
#5 0x562187d6b47f <unknown>
#6 0x562187d62de3 <unknown>
#7 0x562187d382dd <unknown>
#8 0x562187d3934e <unknown>
#9 0x562187f983e4 <unknown>
#10 0x562187f9c3d7 <unknown>
#11 0x562187fa6b20 <unknown>
#12 0x562187f9d023 <unknown>
#13 0x562187f6b1aa <unknown>
#14 0x562187fc16b8 <unknown>
#15 0x562187fc1847 <unknown>
#16 0x562187fd1243 <unknown>
#17 0x718e6ea9ca94 <unknown>
#18 0x718e6eb29c3c <unknown>

Refer to the configuration and troubleshooting guide: https://github.com/feder-cr/LinkedIn_AIHawk_automatic_job_application/blob/main/readme.md#configuration

```

**Comment by talvinder** [comment]
I am facing the same issue. I am running this on my local machine (macos monterey, 12.7.6). Pretty much the same error as well. This seems to be a deeper issue, I have trying to troubleshoot for the last 3-4 hours.

**Comment by talvinder** [comment]
Okay. This is the solution that worked for me.

1. Ensure webdriver_manager is Installed
First, make sure that the webdriver_manager package is installed. This package will automatically download and manage the ChromeDriver binary for you.

`pip install webdriver-manager`

2. Update create_driver_selenium Function
Update the create_driver_selenium function in the lib_resume_builder_AIHawk/utils.py file to use webdriver_manager to install ChromeDriver and specify its path.

Updated Code
```
from selenium import webdriver
from selenium.webdriver.chrome.service import Service as ChromeService
from webdriver_manager.chrome import ChromeDriverManager

def create_driver_selenium():
    options = get_chrome_browser_options()  # Use the method to get Chrome options

    # Use webdriver_manager to install and specify the path to ChromeDriver
    service = ChromeService(executable_path=ChromeDriverManager().install())
    
    return webdriver.Chrome(service=service, options=options)
```

The app ran like butter after this.

**Comment by lscalize** [comment]
@talvinder tks for the help.

I've followed your instructions and faced the same problem.

I think Google Colab is the issue. Which other tool do you suggest? 

![image](https://github.com/user-attachments/assets/fe1c4fa2-04fc-4899-b54f-bb371815f3ec)

**Comment by Lumexio** [comment]
I find the solution to this problem if you are on linux


First of all you need to remove purge uninstall your google chrome installation
Second install back chrome as a package with 
> dpkg i chrome-stable.deb


i explore the code and if you still have problems you can hard-code the chrome installation path as viewed on the screenshot
![image](https://github.com/user-attachments/assets/a092aeba-c10c-4bd4-9108-419833a265dd)

**Comment by lscalize** [comment]
@Lumexio I'm not on linux, but I'm going to try it. Why not? Tks a lot.

**Comment by talvinder** [comment]
> @talvinder tks for the help.
> 
> 
> 
> I've followed your instructions and faced the same problem.
> 
> 
> 
> I think Google Colab is the issue. Which other tool do you suggest? 
> 
> 
> 
> ![image](https://github.com/user-attachments/assets/fe1c4fa2-04fc-4899-b54f-bb371815f3ec)
> 
> 

I don't think this will work on google colab or GitHub codespaces, since it needs a chrome driver (basically a working chrome instance).

**Comment by lscalize** [comment]
@talvinder Tks. So, what would you suggest instead?

**Comment by lscalize** [comment]
I've tried many times and it did not work. 

Could someone help me pls?

**Comment by talvinder** [comment]
> I've tried many times and it did not work. 
> 
> 
> 
> Could someone help me pls?

I ran this on my local machine (MacBook Pro) and with some tinkering it worked fine. The main issue on my local machine was path to chrome driver, once that was sorted, it worked.

To fix the path, I had to make some tweaks in the python package of LinkedInAI. I am forgetting what exactly it was. I will come back with the exact instructions that worked for me once I am in front of my machine.
